### PR TITLE
new wizzard control

### PR DESCRIPTION
**Pull request by dnenov**
### Purpose

A PR containing the preliminary UI work for the design rehaul of the Package Manager publishing package. "Step-by-step wizard. Users should be able to navigate around by clicking the numbers at the top (once each step has been successfully fulfilled)."

The control has not been implemented anywhere in code yet. 

Link to the Figma board: https://www.figma.com/design/MJh76m7bbafE3sjwLUTZhZ/Dynamo-package-manager-update?node-id=2409-15319&t=4BQN0lZMCg2MRHTp-0

### UI Changes

CIGMsb99xD

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [x] This PR modifies some build requirements and the readme is updated
- [x] This PR contains no files larger than 50 MB

### Release Notes

- new wizard-style custom user control

### Reviewers

@QilongTang 
@Amoursol 

### FYIs

@reddyashish

### CODE REVIEW

**Comment by Amoursol** [comment]
Looks AWESOME @dnenov!

**Comment by QilongTang** [comment]
Indeed, looking good man @dnenov !

File: src/DynamoCoreWpf/Views/PackageManager/Controls/StepIndicatorControl.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        {
+            InitializeComponent();
+            this.UpdateVisualState(State);
+            this._clickedDuringMouseOver = false;
+        }
+
+        public static readonly DependencyProperty TitleProperty =
+         DependencyProperty.Register("Title", typeof(string), typeof(StepIndicatorControl), new PropertyMetadata("Step"));
+
+        public string Title
```
Comment (reddyashish) [inline src/DynamoCoreWpf/Views/PackageManager/Controls/StepIndicatorControl.xaml.cs:27]: Add summary to public properties

Comment (dnenov) [reply src/DynamoCoreWpf/Views/PackageManager/Controls/StepIndicatorControl.xaml.cs:33]: Added summaries to all public properties!

**Review by reddyashish — APPROVED** [review]

**Comment by reddyashish** [comment]
Running the job again https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/16406/

**Review by reddyashish — APPROVED** [review]

**Comment by reddyashish** [comment]
passed: https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/16551/

**Comment by QilongTang** [comment]
@reddyashish I think we are supposed to review and merge https://github.com/DynamoDS/Dynamo/pull/15536 instead?

**Comment by reddyashish** [comment]
So this is not needed?

**Comment by reddyashish** [comment]
@QilongTang I can revert this once you confirm it.

**Comment by QilongTang** [comment]
@reddyashish Can you check locally? I think the wizard is not complete, this PR is a WIP. I would prefer we revert in that case

**Pull request merged**